independent-projects/bootstrap/maven-resolver/src/main/java/io/quarkus/bootstrap/resolver/maven/workspace/LocalProject.java

```diff
@@ -30,6 +31,124 @@
     private static final String PROJECT_BASEDIR = "${project.basedir}";
     private static final String POM_XML = "pom.xml";
 
+    private static class WorkspaceLoader {
+
+        private final LocalWorkspace workspace = new LocalWorkspace();
+        private final Map<Path, Model> cachedModels = new HashMap<>();
+        private final Path currentProjectPom;
+        private Path workspaceRootPom;
+
+        private WorkspaceLoader(Path currentProjectPom) throws BootstrapMavenException {
+            this.currentProjectPom = isPom(currentProjectPom) ? currentProjectPom : locateCurrentProjectPom(currentProjectPom, true);
+        }
+
+        private boolean isPom(Path p) {
+            if (Files.exists(p) && !Files.isDirectory(p)) {
+                try {
+                    loadAndCache(p);
+                    return true;
+                } catch (BootstrapMavenException e) {
+                    // not a POM file
+                }
+            }
+            return false;
+        }
+
+        private Model model(Path pomFile) throws BootstrapMavenException {
+            Model model = cachedModels.get(pomFile.getParent());
+            if(model == null) {
+                model = loadAndCache(pomFile);
+            }
+            return model;
+        }
+
+        private Model loadAndCache(Path pomFile) throws BootstrapMavenException {
+            final Model model = readModel(pomFile);
+            cachedModels.put(pomFile.getParent(), model);
+            return model;
+        }
```

Comment:
This snippet can be replaced with: 

```java
        private Model model(Path pomFile) throws BootstrapMavenException {
            return cachedModels.computeIfEmpty(pomFile.getParent(), parent->readModel(pomFile));
        }
```

Reply:
computeIfAbsent()? But then I'd have to catch the exception and re-throw a runtime one.

Reply:
Ah damn checked exceptions, you're right :(